src/linux/index.js

```diff
@@ -54,6 +55,7 @@ const checkifExists = async (protocol) => {
 
 const register = async (options, cb) => {
     let res = null;
+    const tmpDir = fs.mkdtempSync(`${os.tmpdir()}/register-protocol-`);

```

Comment:
looks like a lot of code repetition
lets have a higher-order function wrapping register functions for every os

Reply:
ok, can you move to helper.

Reply:
done, moved to constats